django/contrib/admin/views/autocomplete.py

```diff
@@ -24,12 +24,15 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
+        self.id_field = request.GET.get('id_field', 'pk')
+        # set default id_field to 'pk' if not provided or not to_field_allowed()
+        #self.id_field = self.id_field if self.model_admin.to_field_allowed(request, self.id_field) else 'pk' TODO: not working
         self.paginator_class = self.model_admin.paginator
-        self.object_list = self.get_queryset()
+        self.object_list = self.get_queryset().order_by(self.id_field)
```

Comment:
I don't see a reason for the `order_by` here.

Reply:
I was getting warnings about unordered querysets and thought, to return consistent query results for a given term, to order the results.